addons/account/models/account_account.py

```diff
@@ -344,6 +370,7 @@ def _check_account_is_bank_journal_bank_account(self):
         if self._cr.fetchone():
             raise ValidationError(_("You cannot change the type of an account set as Bank Account on a journal to Receivable or Payable."))
 
+    @api.depends_context('allowed_company_ids')
```

Comment:
Stored fields cannot be company dependent

Reply:
True. I'll need to un-store it (I don't expect this to affect performance much).

Reply:
Okay - I shouldn't unstore it because we need to query this field in DB - instead I'm making it company-dependent and adapting the query accordingly.